components/post_view/post_list_row/post_list_row.jsx

```diff
@@ -23,6 +23,50 @@ export default class PostListRow extends React.PureComponent {
         loadOlderPosts: PropTypes.func,
         loadNewerPosts: PropTypes.func,
         togglePostMenu: PropTypes.func,
+
+        /**
+         * To Check if the current post is last in the list
+         */
+        isLastPost: PropTypes.bool,
+
+        /**
+         * To check if the state of emoji for last message and from where it was emitted
+         */
+        shortcutReactToLastPostEmittedFrom: PropTypes.string,
+
+        actions: PropTypes.shape({
+
+            /**
+             * Function to set or unset emoji picker for last message
+             */
+            emitShortcutReactToLastPostFrom: PropTypes.func
+        }),
+    }
+
+    blockShortcutReactToLastPostForNonMessages(listId) {
+        const {actions: {emitShortcutReactToLastPostFrom}} = this.props;
+        const isDateLine = PostListUtils.isDateLine(listId);
+        const isNewMessageSeparator = PostListUtils.isStartOfNewMessages(listId);
+        const isChannelIntroMessage = listId === PostListRowListIds.CHANNEL_INTRO_MESSAGE;
+        const isLoadMoreButton = listId === PostListRowListIds.LOAD_OLDER_MESSAGES_TRIGGER || listId === PostListRowListIds.LOAD_NEWER_MESSAGES_TRIGGER;
+        const isLoadingScreen = listId === PostListRowListIds.OLDER_MESSAGES_LOADER || listId === PostListRowListIds.NEWER_MESSAGES_LOADER;
+
+        if (isDateLine || isNewMessageSeparator || isChannelIntroMessage || isLoadMoreButton || isLoadingScreen) {
```

Comment:
Can you extract the above condition to be in `post_utils.jsx`? Some of these are part of `isIdNotPost` but i think you can add the rest to the same function and use it.